src/server/game/Spells/SpellInfo.cpp

```diff
@@ -502,10 +502,13 @@ int32 SpellEffectInfo::CalcValue(Unit const* caster, int32 const* bp, Unit const
 
             if (canEffectScale)
             {
-                GtNPCManaCostScalerEntry const* spellScaler = sGtNPCManaCostScalerStore.LookupEntry(_spellInfo->SpellLevel - 1);
-                GtNPCManaCostScalerEntry const* casterScaler = sGtNPCManaCostScalerStore.LookupEntry(caster->GetLevel() - 1);
-                if (spellScaler && casterScaler)
-                    value *= casterScaler->ratio / spellScaler->ratio;
+                CreatureTemplate const* cInfo = caster->ToCreature()->GetCreatureTemplate();
+
```

Comment:
Is there 100% no risk of this ever being null?

Reply:
> Is there 100% no risk of this ever being null?

Caster is evaluated on L450, and, correct me if I'm wrong, I don't see either method being at risk of causing a null given caster exists.

Reply:
Yeah this doesn't look like it could be null to me. Maybe if caster somehow can't be cast to Creature

Reply:
I did not know caster was already checked. Then no, the only way it would null is of there is no ct, which is highly unlikely anyway

Reply:
It’s not granted to be a creature, and in that case, it will crash